File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/RelatedPerson.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Address
from base import Attachment
from base import Period
from base import CodeableConcept
from base import ContactPoint
from base import HumanName
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class RelatedPersonCommunication(BackboneElement):
    language: CodeableConcept
    preferred: Optional[bool] = None

@dataclass(kw_only=True)
class RelatedPerson(DomainResource):
    patient: Reference
    address: Optional[List[Address]] = field(default_factory=list)
    name: Optional[List[HumanName]] = field(default_factory=list)
    birth_date: Optional[str] = None
    relationship: Optional[List[CodeableConcept]] = field(default_factory=list)
    photo: Optional[List[Attachment]] = field(default_factory=list)
    active: Optional[bool] = None
    communication: Optional[List[RelatedPersonCommunication]] = field(default_factory=list)
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    telecom: Optional[List[ContactPoint]] = field(default_factory=list)
    gender: Optional[str] = None
    period: Optional[Period] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/RelatedPerson.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class RelatedPerson : DomainResource, IResource
{
    public required Base.ResourceReference Patient { get; set; }
    public Base.Address[]? Address { get; set; }
    public Base.HumanName[]? Name { get; set; }
    public string? BirthDate { get; set; }
    public Base.CodeableConcept[]? Relationship { get; set; }
    public Base.Attachment[]? Photo { get; set; }
    public bool? Active { get; set; }
    public RelatedPerson_Communication[]? Communication { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ContactPoint[]? Telecom { get; set; }
    public string? Gender { get; set; }
    public Base.Period? Period { get; set; }

    public class RelatedPerson_Communication : BackboneElement
    {
        public required Base.CodeableConcept Language { get; set; }
        public bool? Preferred { get; set; }
    }
}